### PR TITLE
Clean up #35514

### PR DESCRIPTION
**Pull request by rallytime**
### What does this PR do?

This updates the test file to maintain two different tests for Py2 vs 3. Thanks for the suggestion @ticosax.
### What issues does this PR fix or reference?

Fixes up saltstack/salt#35514 according to the suggestions so we can get that change in.
### Previous Behavior

Remove this section if not relevant
### New Behavior

Remove this section if not relevant
### Tests written?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Comment by rallytime** [comment]
After thinking about this some more, I realized that we will run into a problem on Python 2.6, even with the different `skipIfs` for the Python2 and Python3 specific tests. Instead of maintaining two tests, lets just used nested with statements. That way the test will work the same on Python 2.6, 2.7, and 3.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Hrm. I am not sure why changing one test is causing another one to fail here. I can't reproduce it locally either. I'll have to take a closer look next week.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
This is still on my radar. Hopefully I will be able to look at this test failure tomorrow after rebasing it on the latest develop today.

File: salt/utils/thin.py [hunk 1]
```diff
@@ -269,6 +269,8 @@ def gen_thin(cachedir, extra_mods='', overwrite=False, so_mods='',
     tempdir = None
     for py_ver, tops in six.iteritems(tops_py_version_mapping):
         for top in tops:
+            if not os.path.isabs(top):
+                continue
```
Comment (bbinet) [inline salt/utils/thin.py]: Sorry, what was the reasoning behind this change?

Personally, I'm used to generate salt thin tarballs with local python packages, and it was working perfectly until this change. If it is not possible to remove this change, would it be possible to add an `absonly=False` arg to enable support for non absolute packages?

Comment (bbinet) [inline salt/utils/thin.py]: @cachedout @rallytime ?
Should I propose a pull request to make it configurable?

Comment (rallytime) [inline salt/utils/thin.py]: @bbinet Yes, I think that would be a good approach to make it configurable.

Comment (bbinet) [inline salt/utils/thin.py]: See PR #39415

**Pull request merged**